WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -248,19 +248,43 @@ class PostCoordinator: NSObject {
 
 extension PostCoordinator: Uploader {
     func resume() {
-        let service = PostService(managedObjectContext: mainContext)
-
-        service.getFailedPosts { [weak self] posts in
+        let fetcher = FailedPostsFetcher(mainContext)
+        fetcher.getPostsToRetry { [weak self] posts in
             guard let self = self else {
                 return
             }
 
-            posts.forEach() { post in
-                let shouldRetry = post.status == .draft && !post.hasRemote()
+            posts.forEach({ self.retrySave(of: $0) })
+        }
+    }
+}
 
-                if shouldRetry {
-                    self.retrySave(of: post)
-                }
+extension PostCoordinator {
+    /// Fetches failed posts that should be retried when there is an internet connection.
+    class FailedPostsFetcher {
+        private let postService: PostService
+
+        init(_ postService: PostService) {
+            self.postService = postService
+        }
+
+        init(_ managedObjectContext: NSManagedObjectContext) {
+            postService = PostService(managedObjectContext: managedObjectContext)
+        }
+
+        func getPostsToRetry(result: @escaping ([AbstractPost]) -> Void) {
+            let allowedStatuses: [BasePost.Status] = [.draft, .publish]
```

Comment:
should this include `.publishPrivate`? if not, why not?

Reply:
Instead of fully implementing #12227, I wanted to limit the scope to just #12240 which will only support published posts for now. There are only a few steps towards supporting `.publishPrivate` like adding a different message for the Snack bar and Post List:

> Private post will be published next time your device is online

Instead of: 

> Post will be published next time your device is online

Not adding `.publishPrivate` for now decreases the amount of code which decreases the amount of testing we will need to do in order to achieve the full workflow. 

<img src="https://user-images.githubusercontent.com/198826/62169329-2e90b080-b2e5-11e9-9ca4-203eaeab6e71.png">

I have listed the known tasks in #12240. 

What do you think about this plan, @jklausa?